pkg/service/roommanager.go

```diff
@@ -317,40 +317,55 @@ func (r *RoomManager) getOrCreateRoom(ctx context.Context, roomName livekit.Room
 		return nil, err
 	}
 
-	// construct ice servers
-	room = rtc.NewRoom(ri, *r.rtcConfig, &r.config.Audio, r.telemetry)
-	room.Hold()
+	r.lock.Lock()
+
+	currentRoom := r.rooms[roomName]
+	for currentRoom != lastSeenRoom {
```

Comment:
if hold fails, I think we should let it flow through and create a new room. we want to avoid an infinite loop here in case hold keeps failing.

Reply:
You're a super fast reviewer! =D

Hmm, in which case hold would keep failing? I don't think it has something to do with network. If this is completely local, can we just assume it will succeed in few attempts?

I had to add a loop, otherwise we once again fall into the possibility of creating duplicate rtc room.

Reply:
If the room has already been closed, `Hold` would return false. In this case, it'll never come back, and creating a new room is the only way to go here.

Reply:
This code covers that case. A new room will be created, and the rest will pick up that room if not closed.

Reply:
true, you are right!